### DISCUSSION TITLE
Dioxus full stack Allow a server main function

### DISCUSSION THREAD

**Discussion by enricoschaaf** [DioxusLabs/dioxus#3010] [Ideas]
Hi 👋🏽 

To be able to run more complicated full stack applications with dioxus it would be really nice if we could provide our own main function when doing fullstack and then setting up the backend side of dioxus in our main function ourselves.

What do you think?

**Comment by wiseaidev** [comment]
I'm not sure if I understood the question correctly, but Dioxus allows you to customize both the client and server according to your needs. You can check out [this example](https://github.com/opensass/aibook/blob/main/src/main.rs) for more details.
